netdcli/netdcli.go

```diff
@@ -373,46 +385,43 @@ func executeOpts(opts *cliOpts) error {
 
 	err := validateOpts(opts)
 	if err != nil {
-		log.Fatalf("error %v validating opts\n", err)
+		return err
 	}
 
-	etcdDriver := &state.EtcdStateDriver{}
-	driverConfig := &state.EtcdStateDriverConfig{}
-	driverConfig.Etcd.Machines = []string{opts.etcdURL}
-	config := &core.Config{V: driverConfig}
-	err = etcdDriver.Init(config)
+	stateDriver, err := utils.GetStateDriver()
```

Comment:
`GetStateDriver()` is just a utility to return a previously instantiated singleton state-driver (through a call to `NewStateDriver()`) so it doesn't need to take any arguments. I process the connection flags and init the state-driver instance in netdcli in the `initStateDriver()` function that is called just before `executeOpts()` and `executeJson()`